crates/rocksdb/src/lib.rs

```diff
@@ -514,7 +515,7 @@ where
                 }
                 return result;
             }
-            _ = tokio::time::sleep(manager.stall_detection_duration()), if !stalled => {
+            _ = &mut timeout, if !stalled => {
```

Comment:
We are trying to save the costs to create one additional `tokio::time::sleep::Sleep` after it has fired, right? Did you see the costs of this operation somewhere?

Reply:
It showed up as an offender in my profiles, every bit helps :)